clash-lib/src/Clash/Normalize/Transformations.hs

```diff
@@ -2391,3 +2392,78 @@ separateArguments (TransformContext is0 _) e@(collectArgsTicks -> (Var g, args,
         return [(ty,arg)]
 
 separateArguments _ e = return e
+
+xOptimize :: HasCallStack => NormRewrite
+xOptimize (TransformContext is0 _) e@(Case subj ty alts) =
+  case List.partition (isUndefined . snd) alts of
+    ([], _)    -> return e
+    (_, [])    -> changed (Prim "Clash.XException.errorX" (PrimInfo ty WorkConstant))
```

Comment:
The reasoning is that me and Leon looked at it and went "what should it be, maybe Constant"?
If `WorkNone` fits better that's no problem

Reply:
Right. `WorkConstant` means that the whole primitive is constant (i.e., does not perform any work) if it's arguments are constant. But the whole thing gets replaced by X, so it will never perform any work. Hence `WorkNever` should do.